Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/ProcessWeightTests.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BuildXL.Scheduler.Distribution;
+using Test.BuildXL.Executables.TestProcess;
+using Test.BuildXL.Scheduler;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTest.BuildXL.Scheduler
+{
+    public class ProcessWeightTests : SchedulerIntegrationTestBase
+    {
+        public ProcessWeightTests(ITestOutputHelper output) : base(output)
+        {
+            // Turns on logging for most scheduler stats, including the limiting resource stat looked for by these tests
+            ShouldLogSchedulerStats = true;
+        }
+
+        [Theory]
+        [InlineData(-1)] // Make sure a weights of < 1 are treated the same as a weight of 1
+        [InlineData(0)]
+        [InlineData(1)]
+        public void CanRunUpToMaxProcessesInParallel(int defaultWeight)
+        {
+            CreateAndScheduleProcessWithWeight(defaultWeight, numProcesses: Configuration.Schedule.MaxProcesses);

```

Comment:
>MaxProcesses [](start = 99, length = 12)

Does the test configuration set MaxProcesses or does it flow in based on the machine's core count? I'd suggest pinning in here in the class since the tests depend on its value

Reply:
All these tests do things relative to MaxProcesses value set, so it should be fine if the value fluctuates based off machine.

Manually setting the configuration does trump the machine's configuration and I have a test for that below with "TotalAllowedWeightIsLimitedByConfigurationMaxProcesses"



---
In reply to: [271088728](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271088728) [](ancestors = 271088728)

Reply:
It'd be a waste for these tests to spawn 80 child processes when run on powerful machines. I also worry about variability in the tests making them more error prone or harder to debug

Reply:
Fair, I can limit them to some constant.

---
In reply to: [271091494](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271091494) [](ancestors = 271091494)